src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/ImmutableByteArrayInterop.cs

```diff
@@ -84,7 +84,7 @@ internal static byte[] DangerousGetUnderlyingArray(ImmutableArray<byte> array)
         {
             ByteArrayUnion union = default;
             union.ImmutableArray = array;
-            return union.UnderlyingArray;
+            return union.UnderlyingArray!;
```

Comment:
From doc above the method could return null, also seems `union.UnderlyingArray` should be nullable too

> /// \<returns>The underlying array, or null if \<see cref="ImmutableArray{T}.IsDefault"/> is true.\</returns>